ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js

```diff
@@ -66,7 +66,9 @@ module.exports = Marionette.LayoutView.extend({
       this.model.get('filterTree') !== undefined
     ) {
       this.queryAdvanced.currentView.deserialize(
-        cql.simplify(JSON.parse(this.model.get('filterTree')))
+        typeof this.model.get('filterTree') === 'string'
+          ? JSON.parse(this.model.get('filterTree'))
+          : this.model.get('filterTree')
```

Comment:
What does this do?

Reply:
Sometimes `this.model.get('filterTree')` is already a JSON, which causes issues when you try to turn it into a JSON again. But sometimes it's a string, which needs to be converted to a JSON. This ternary avoids issues stemming from those inconsistencies